validation-api/azure_utils/auth.py

```diff
@@ -15,29 +15,24 @@ def read_config(config_path):
     return config
 
 
-def obtain_credential(config, credential_type="default"):
-    """Obtains client credentials from Azure KeyVault.
-    Args:
-        config (dict): Dictionary of configuration values.
+def obtain_sp_credential():
+    """Obtains service principal credentials from Azure.
     Returns:
-        Instance of ClientSecretCredential.
+        Instance of DefaultAzureCredential.
```

Comment:
What's the main difference between `ClientSecretCredential` and `DefaultAzureCredential`?

Reply:
`DefaultAzureCredential` will basically check a chain of different authentication methods and return the first that succeeds (in this case, it checks the env variables). This way, if we deploy an app and use a different authentication method (eg Managed Identity or something), then this would still work. `ClientSecretCredential` only works with a specified client secret so if we changed how we authenticate then we would have to change that as well. There's a good overview here: https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python?tabs=managed-identity%2Croles-azure-portal%2Csign-in-azure-cli&pivots=blob-storage-quickstart-scratch